cmd/infra/powervs/destroy.go

```diff
@@ -740,3 +766,189 @@ func destroyVpcLB(ctx context.Context, options *DestroyInfraOptions, subnetID st
 
 	return pagingHelper(f)
 }
+
+// createCOSClient creates COS client to interact with the COS for clean up
+func createCOSClient(serviceInstanceCRN string, location string) (*s3.S3, error) {
+	// if IBMCLOUD_COS_API_ENDPOINT is set, will use custom endpoint with the default cosEndpoint
+	serviceEndpoint := getCustomEndpointUrl(cosService, fmt.Sprintf("s3.%s.cloud-object-storage.appdomain.cloud", location))
+
+	awsOptions := cosSession.Options{
+		Config: aws.Config{
+			Endpoint: &serviceEndpoint,
+			Region:   &location,
+			HTTPClient: &http.Client{
+				Transport: &http.Transport{
+					Proxy: func(req *http.Request) (*url.URL, error) {
+						return httpproxy.FromEnvironment().ProxyFunc()(req.URL)
+					},
+					DialContext: (&net.Dialer{
+						Timeout:   30 * time.Second,
+						KeepAlive: 30 * time.Second,
+						DualStack: true,
+					}).DialContext,
+					ForceAttemptHTTP2:     true,
+					MaxIdleConns:          100,
+					IdleConnTimeout:       90 * time.Second,
+					TLSHandshakeTimeout:   10 * time.Second,
+					ExpectContinueTimeout: 1 * time.Second,
+				},
+			},
+			S3ForcePathStyle: aws.Bool(true),
+		},
+	}
+
+	awsOptions.Config.Credentials = ibmiam.NewStaticCredentials(aws.NewConfig(), getCustomEndpointUrl(platformService, iamEndpoint), cloudApiKey, serviceInstanceCRN)
+
+	sess, err := cosSession.NewSessionWithOptions(awsOptions)
+	if err != nil {
+		return nil, err
+	}
+	return s3.New(sess), nil
+}
+
+// findCOSInstance find COS resource instance by name
+func findCOSInstance(rcv2 *resourcecontrollerv2.ResourceControllerV2, cosInstanceName string, resourceGroupID string) (*resourcecontrollerv2.ResourceInstance, error) {
+	instances, resp, err := rcv2.ListResourceInstances(
+		&resourcecontrollerv2.ListResourceInstancesOptions{
+			Name:            &cosInstanceName,
+			ResourceGroupID: &resourceGroupID,
+			ResourcePlanID:  utilpointer.String("744bfc56-d12c-4866-88d5-dac9139e0e5d"),
```

Comment:
where is this magic number coming from? It'd be nice to add a comment clarifying

Reply:
Added a comment